airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -64,18 +63,15 @@ def use_cache(self) -> bool:
         return False
 
     def request_cache(self) -> requests.Session:
-        self.clear_cache()
-        return requests_cache.CachedSession(self.cache_filename)
+        backend = requests_cache.SQLiteCache(use_temp=True)
+        return requests_cache.CachedSession(self.cache_filename, backend=backend)
 
     def clear_cache(self) -> None:
         """
-        remove cache file only once
+        clear cached requests for current session, can be called any time
         """
-        STREAM_CACHE_FILES = globals().setdefault("STREAM_CACHE_FILES", set())
-        if self.cache_filename not in STREAM_CACHE_FILES:
-            with suppress(FileNotFoundError):
-                os.remove(self.cache_filename)
-            STREAM_CACHE_FILES.add(self.cache_filename)
+        if isinstance(self._session, requests_cache.CachedSession):
+            self._session.cache.clear()
```

Comment:
@keu I'm a little worried about this implementation because initializing a `Stream` will clear it's cache.

I think this would be fine if all streams were initialized at the start of a sync, but some connectors instantiate their streams while the sync is in progress. [source-zenloop is an example of such connector](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-zenloop/source_zenloop/streams.py#L73-L75).

The problem I'm picturing is we'll read data for the parent stream, then we'll clear the cache when generating the stream_slices for the substream, making the cache useless...

I suspect that's why the existing implementation does not remove the file if the filename is in the global set

Reply:
@girarda you are totally right, I think this was one of the main reason, but the implementation is so implicit. I will try to find better solution, but not going to spend much time, in the worst case will just revert this STREAM_CACHE_FILES.

Reply:
I think if I can prove that we have a new file each time we start sync then I can just remove this call `clear_cache` from the `__init__`

Reply:
@girarda, here is the explanation of how tempdir works https://docs.python.org/3/library/tempfile.html#tempfile.tempdir
I can say that tempdir behaves like a singleton and generates a new folder each run.

Reply:
@keu are the files deleted after the run?

Reply:
@girarda generally speaking, tempdirs and tempfiles are not deleted when the Python program finishes. It is just that files placed in the OS $TEMP folder will be cleaned later (depending on the type of OS, after reboot, or periodically after some time). We could enforce cleaning of these folders/files using a contextmanager (TempDir, TempFile) but it  means we need to find the right scope for a context (like source level, or main.py level).

Reply:
@keu I think entrypoint.py would be a good place for the context manager since [it's already managing temporary files](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/entrypoint.py#L96)

Reply:
@girarda  This is great! Please check my updates.